src/EFCoreTestAppWithChronicleSaga/Handlers/OrderSagaHandler.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Chronicle;
+using MediatR;
+using EFCoreTestApp.Commands;
+using EFCoreTestApp.Events;
+using EFCoreTestApp.Persistence;
+
+namespace EFCoreTestApp.Handlers
+{
+    public class OrderSagaHandler: IRequestHandler<CreateOrder>, INotificationHandler<OrderCreated>
+    {
+        /*
+            NOTE: Remove ISagaUnitOfWork if the Chronicles Internal EFCore/SQL Server implementation is being used.
+            If implementing with a custom Solution enable it.
+        */
+
+        private readonly ISagaCoordinator _coordinator;
+        private ICustomUnitOfWork<SagaDbContext> SagaUnitOfWork { get; }
+
+        public OrderSagaHandler(ISagaCoordinator coordinator, ICustomUnitOfWork<SagaDbContext> _sagaUnitOfWork)
+        {
+            _coordinator = coordinator;
+            SagaUnitOfWork = _sagaUnitOfWork;
+        }
+
+        public async Task<Unit> Handle(CreateOrder command, CancellationToken cancellationToken)
+        {
+            await _coordinator.ProcessAsync(command, SagaContext.Empty);
+            // once every thing is processed in the handler, commit changes to DB.
+            // This can be moved else where depending on the user needs.
+            await SagaUnitOfWork.CommitAsync();
```

Comment:
Honestly, I don't get why to explicitly manage UoW. To me the coordinator should be responsible for committing the entire TX to DB, otherwise the whole API is not consistent for different DB providers.

Reply:
The idea was to commit everything as one unit. Since [here](https://github.com/snatch-dev/Chronicle/blob/1c414d4714af3781bcfba9ee6d57d3c5f5f465e9/src/Chronicle/Managers/SagaProcessor.cs#L58) I see that log and state are entered separately to DB.  
My idea was to create a unit of work and expose that to the client application so that they could manage their data & changes along with the state as part of one Atomic Operation (and mention that in the documentation how to use it).  
But you are absolutely correct that it makes the overall API inconsistent with other Providers.  
I will then change the code to remove the unit of work from there.  
Anyways I already have one [example](https://github.com/UmerIftikhar/Chronicle/tree/master/src/EFCoreTestAppWithCutomSaga/Persistence) of creating Custom Saga (with EFCore), and the client can use that if they want more control.

Reply:
If we register the DbContext as Transient, then we can not wrap the following code inside a transaction? right?
![image](https://user-images.githubusercontent.com/11350146/98467641-0fb6e480-21df-11eb-9cf6-3176b5ce1a97.png)

And if we register the DbContext as Scoped, then we will face this error:

> A second operation started in this context before a previous operation completed. This is usually caused by different threads using the same instance of DbContext.